src/main/java/com/scalar/cassy/transferer/AzureBlobFileUploader.java

```diff
@@ -40,57 +46,83 @@ public AzureBlobFileUploader(BlobContainerAsyncClient blobContainerClient) {
     }
 
     logger.info("Uploading " + file);
-    return blobContainerClient
-        .getBlobAsyncClient(key)
-        .uploadFromFile(file.toString(), true)
-        .toFuture();
+    return executorService.submit(
+        () -> {
+          try {
+            try (InputStream inputStream = new FileInputStream(file.toString())) {
+              blobContainerClient
+                  .getBlobClient(key)
+                  .upload(inputStream, new File(file.toString()).length(), true);
+            }
+          } catch (IOException e) {
+            throw new FileTransferException(e);
+          }
+
+          logger.info("Upload file succeeded : " + file.toString());
+          return null;
+        });
   }
 
   @Override
   public void upload(List<Path> files, BackupConfig config) {
     Path dataDir = Paths.get(config.getDataDir());
-    List<Mono<Void>> toBeUploaded = new ArrayList<>();
+    List<Future> uploads = new ArrayList<>();
```

Comment:
```suggestion
    List<Future<Void>> uploads = new ArrayList<>();
```

Reply:
@brfrn169 san,
Sorry I don't get why we use `List<Future<Void>>` instead of `List<Future>`.

Reply:
`Future` is a generic type, and using a generic type without a type parameter causes a warning. So we shouldn't always omit the type parameter.

Reply:
Thank you for your suggestion and explanation.
Suggested changes integrated f8d33d6